tests/Fixer/FunctionNotation/PhpdocToReturnTypeFixerTest.php

```diff
@@ -0,0 +1,200 @@
+            ],
+            'fix scalar types when configured' => [
+                '<?php /** @return int */ function my_foo(): int {}',
+                '<?php /** @return int */ function my_foo() {}',
+                null,
+                ['scalar_types' => true],
+            ],
+            'array native type' => [
+                '<?php /** @return array */ function my_foo(): array {}',
+                '<?php /** @return array */ function my_foo() {}',
+            ],
+            'callable type' => [
+                '<?php /** @return callable */ function my_foo(): callable {}',
+                '<?php /** @return callable */ function my_foo() {}',
+            ],
+            'self accessor' => [
+                '<?php
+                    class Foo {
+                        /** @return self */ function my_foo(): self {}
+                    }
+                ',
+                '<?php
+                    class Foo {
+                        /** @return self */ function my_foo() {}
+                    }
+                ',
+            ],
+            'report static as self' => [
+                '<?php
+                    class Foo {
+                        /** @return static */ function my_foo(): self {}
+                    }
+                ',
+                '<?php
+                    class Foo {
+                        /** @return static */ function my_foo() {}
+                    }
+                ',
+            ],
+            'skip resource special type' => [
+                '<?php /** @return resource */ function my_foo() {}',
+            ],
+            'skip mixed special type' => [
+                '<?php /** @return mixed */ function my_foo() {}',
+            ],
+            'null alone cannot be a return type' => [
+                '<?php /** @return null */ function my_foo() {}',
+            ],
+            'skip mixed types' => [
+                '<?php /** @return Foo|Bar */ function my_foo() {}',
+            ],
+            'nullable type' => [
+                '<?php /** @return null|Bar */ function my_foo(): ?Bar {}',
+                '<?php /** @return null|Bar */ function my_foo() {}',
+                70100,
+            ],
+            'nullable type reverse order' => [
+                '<?php /** @return Bar|null */ function my_foo(): ?Bar {}',
+                '<?php /** @return Bar|null */ function my_foo() {}',
+                70100,
+            ],
+            'nullable native type' => [
+                '<?php /** @return null|array */ function my_foo(): ?array {}',
+                '<?php /** @return null|array */ function my_foo() {}',
+                70100,
+            ],
+            'skip mixed nullable types' => [
+                '<?php /** @return null|Foo|Bar */ function my_foo() {}',
+            ],
+            'skip generics' => [
+                '<?php /** @return array<int, bool> */ function my_foo() {}',
```

Comment:
Why ignoring those? An `array` return type would be valid wouldn't it?

Reply:
Indeed it would be.
I'm just not confortable enough supporting something that isn't widely accepted.

Reply:
Generics syntax is not widely used indeed, but `Foo[]` syntax on the other hand is very common, I think it would be nice to at least support this one.

Reply:
see https://github.com/FriendsOfPHP/PHP-CS-Fixer/pull/3640#discussion_r177973321